File: test/resources/HelloProphecy/prophecy_scala/pipelines/silver_zip_codes/code/src/main/scala/io/prophecy/pipelines/silver_zip_codes/graph/CastDataTypes.scala
package io.prophecy.pipelines.silver_zip_codes.graph

import io.prophecy.libs._
import io.prophecy.pipelines.silver_zip_codes.udfs.PipelineInitCode._
import io.prophecy.pipelines.silver_zip_codes.udfs.UDFs._
import io.prophecy.pipelines.silver_zip_codes.config.Context
import org.apache.spark._
import org.apache.spark.sql._
import org.apache.spark.sql.functions._
import org.apache.spark.sql.types._
import org.apache.spark.sql.expressions._
import java.time._

object CastDataTypes {

  def apply(context: Context, in: DataFrame): DataFrame =
    in.select(col("zipcode"),
              col("agi_stub").cast(IntegerType).as("income_bracket"),
              col("N1").cast(IntegerType).as("num_returns"),
              col("STATE").as("state")
    )

}

File: test/resources/HelloProphecy/prophecy/pipelines/silver_zip_codes/code/silver_zip_codes/graph/CastDataTypes.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.utils import *
from prophecy.libs import typed_lit
from silver_zip_codes.config.ConfigStore import *
from silver_zip_codes.udfs.UDFs import *

def CastDataTypes(spark: SparkSession, in0: DataFrame) -> DataFrame:
    return in0.select(
        col("zipcode"), 
        col("agi_stub").cast(IntegerType()).alias("income_bracket"), 
        col("N1").cast(IntegerType()).alias("num_returns"), 
        col("STATE").alias("state")
    )
